smallvil/src/handlers/compositor.rs

```diff
@@ -40,6 +43,25 @@ impl CompositorHandler for Smallvil {
 
         xdg_shell::handle_commit(&self.space, surface);
         resize_grab::handle_commit(&mut self.space, surface);
+
+        self.popups.commit(surface);
+        if let Some(popup) = self.popups.find_popup(surface) {
+            let PopupKind::Xdg(ref popup) = popup;
+            let initial_configure_sent = with_states(surface, |states| {
+                states
+                    .data_map
+                    .get::<XdgPopupSurfaceData>()
+                    .unwrap()
+                    .lock()
+                    .unwrap()
+                    .initial_configure_sent
+            });
+            if !initial_configure_sent {
+                // NOTE: This should never fail as the initial configure is always
+                // allowed.
+                popup.send_configure().expect("initial configure failed");
+            }
+        }
```

Comment:
Could we move this to xdg specific commit handler? Aka `xdg_shell::handle_commit` from line 44
My idea was to have everything xdg related in a single xdg file, where both commit and xdg handler logic lives

Reply:
Done.